src/execution_plan/execution_plan_build/execution_plan_construct.c

```diff
@@ -137,13 +137,14 @@ void ExecutionPlan_PlaceFilterOps(ExecutionPlan *plan, OpBase *root, const OpBas
 	 * after which the filter tree can be applied. */
 	uint nfilters = array_len(sub_trees);
 	for(uint i = 0; i < nfilters; i++) {
-		FT_FilterNode *tree = sub_trees[i];
+		FT_FilterNode *tree = FilterTree_Clone(sub_trees[i]);
```

Comment:
I don't think we need to clone, simply pass `sub_trees[i]` to `NewFilterOp` this can also save us from calling `FilterTree_Free(ft);` at the very end.